core/src/main/java/nl/nn/adapterframework/parameters/Parameter.java

```diff
@@ -577,6 +509,89 @@ public Object getValue(ParameterValueList alreadyResolvedParameters, Message mes
 		return result; 
 	}
 
+	/** Converts raw data to configured parameter type */
+	private Object getTypeValue(Object message, boolean namespaceAware) throws ParameterException {
+		Object result = message;
+		if(!(result instanceof String)) {
+			try {
+				result = Message.asMessage(result).asString();
+			} catch (IOException e) {
+				throw new ParameterException("Could not convert parameter ["+getName()+"] to string", e);
+			}
+		}
```

Comment:
It is a bit sorry to convert all messages to string here, and then later convert them into something else.
Please use message as the main type here.

Reply:
To further elaborate Gerrit's comment here, the `XmlUtils.buildDomDocument` method can also take a Reader for example. Instead of (many) conversions from binary data to character data (or whatever) it can also be directly parsed with the correct reader, improving performance.
If you were to create a new method to convert the Message to a String, you would also no longer need the casts `(String) result` nor `result.toString()` when parsing the value.